Apps/W1/EDocumentConnectors/SignUp/app/src/Processing.Codeunit.al

```diff
@@ -0,0 +1,573 @@
+        exit(true);
+    end;
+
+    local procedure GetNumberOfReceivedDocuments(InputTxt: Text): Integer
+    var
+        JsonManagement: Codeunit "JSON Management";
+        Value: Text;
+    begin
+        InputTxt := LeaveJustNewLine(InputTxt);
+
+        if not JsonManagement.InitializeFromString(InputTxt) then
+            exit(0);
+
+        JsonManagement.GetArrayPropertyValueAsStringByName('inbox', Value);
+        JsonManagement.InitializeCollection(Value);
+
+        exit(JsonManagement.GetCollectionCount());
+    end;
+
+    local procedure CheckIfDocumentStatusSuccessful(EDocument: Record "E-Document"; var HttpRequestMessage: HttpRequestMessage; var HttpResponse: HttpResponseMessage): Boolean
+    var
+        ErrorDescription: Text;
+    begin
+        if not Connection.CheckDocumentStatus(EDocument, HttpRequestMessage, HttpResponse, true) then
+            exit(false);
+
+        if DocumentHasErrorOrProcessing(EDocument, HttpResponse, ErrorDescription) then
+            exit(false);
+
+        exit(true);
+    end;
+
+    local procedure ParseSendFileResponse(HttpContentResponse: HttpContent): Text
+    var
+        JsonManagement: Codeunit "JSON Management";
+        Result: Text;
+        Value: Text;
+    begin
+        Result := Helpers.ParseJsonString(HttpContentResponse);
+        if Result = '' then
+            exit('');
+
+        if not JsonManagement.InitializeFromString(Result) then
+            exit('');
+
+        JsonManagement.GetStringPropertyValueByName('peppolInstanceId', Value);
+        exit(Value);
+    end;
+
+    local procedure SetEDocumentFileID(EDocEntryNo: Integer; FileId: Text)
+    var
+        EDocument: Record "E-Document";
+    begin
+        if FileId = '' then
+            exit;
+        if not EDocument.Get(EDocEntryNo) then
+            exit;
+
+        EDocument."Document Id" := CopyStr(FileId, 1, MaxStrLen(EDocument."Document Id"));
+        EDocument.Modify();
+    end;
+
+    local procedure DocumentHasErrorOrProcessing(EDocument: Record "E-Document"; HttpResponse: HttpResponseMessage; var ErrorDescription: Text): Boolean
+    var
+        EDocumentService: Record "E-Document Service";
+        EDocumentServiceStatus: Record "E-Document Service Status";
+        JsonManagement: Codeunit "JSON Management";
+        GetReadyStatus: Codeunit GetReadyStatus;
+        BlankRecordId: RecordId;
+        HttpContentResponse: HttpContent;
+        Result, Value : Text;
+    begin
+        HttpContentResponse := HttpResponse.Content;
+        Result := Helpers.ParseJsonString(HttpContentResponse);
+        if Result = '' then
+            exit(true);
+
+        if not JsonManagement.InitializeFromString(Result) then
+            exit(true);
+
+        JsonManagement.GetArrayPropertyValueAsStringByName('status', Value);
+
+        if Value in ['Sent'] then begin
+            GetReadyStatus.ScheduleEDocumentJob(Codeunit::GetReadyStatus, BlankRecordId, 120000);
```

Comment:
Couple of things:
1. Dont schedule jobs from a function called "HasSomething". It should check a state. 
2. Question @geschwint What is the flow here? Why cant you rely on the GetResponse job queue job to run? 
    - Want to understand the logic here.